src/Compilers/Core/Portable/Operations/BasicBlock.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    /// <summary>
+    /// PROTOTYPE(dataflow): Add documentation
+    /// </summary>
+    public enum BasicBlockKind
+    {
+        Entry,
+        Exit,
+        Block
+    }
+
+    /// <summary>
+    /// PROTOTYPE(dataflow): Add documentation
+    /// </summary>
+    public sealed class BasicBlock
+    {
+        private ImmutableArray<IOperation>.Builder _statements;
+        private ImmutableHashSet<BasicBlock>.Builder _predecessors;
+
+        public BasicBlock(BasicBlockKind kind)
+        {
+            Kind = kind;
+            _statements = ImmutableArray.CreateBuilder<IOperation>();
+            _predecessors = ImmutableHashSet.CreateBuilder<BasicBlock>();
+        }
+
+        public BasicBlockKind Kind { get; private set; }
+        public ImmutableArray<IOperation> Statements => _statements.ToImmutable();
```

Comment:
Could `BasicBlock` be split into an `internal` builder and a `public` immutable type?

Reply:
Added a comment

---
In reply to: [162214849](https://github.com/dotnet/roslyn/pull/24263#discussion_r162214849) [](ancestors = 162214849)